data/block/blockV2.go

```diff
@@ -629,3 +630,25 @@ func (hv2 *HeaderV2) GetAdditionalData() headerVersionData.HeaderAdditionalData
 	}
 	return additionalVersionData
 }
+
+// CheckFieldsForNil checks a predefined set of fields for nil values
+func (hv2 *HeaderV2) CheckFieldsForNil() error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	err := hv2.Header.CheckFieldsForNil()
+	if err != nil {
+		return err
+	}
+	if hv2.ScheduledRootHash == nil {
```

Comment:
All the fields checked here of the genesis block are not nil?

Reply:
we do not have genesis block on the version 2 of the block

Reply:
Yep, ok